### DISCUSSION TITLE
How to enable feed extension for blog posts only?

### DISCUSSION THREAD

**Discussion by Sussshhhh** [datenstrom/community#914] [Ask a question] [answered]
Hello,

Thanks for Yellow.

I was wondering if it is possible to _only_ include _blog posts_ in the feed without using the tag-, folder-, author-filter option? I found an old blog post suggesting `FeedFilter: blog` in the _config.ini_, but adding this to _yellow-system.ini_ didn't bring the right result.

Thanks in advance!

Best

**Comment by schulle4u** [comment]
Currently the `folder` argument is the only way to limit feed entries to certain locations, e. g. blog. The `FeedFilter` or `FeedFilterLayout` setting unfortunately has been removed in favor of the new argument.

**Reply by Sussshhhh** [reply]
Thanks for your fast and friendly reply. The removal of the mentioned settings is really unfortunate indeed; For blogging purposes it would be great to be able to adapt the main feed to that need. If one wants to exclude pages from the feed, but list them in the navigation, I can't think of a `Status:` workaround either. Thanks again.

**Reply by Sussshhhh** [reply]
For now I have hidden the navigation with `display:none` in the theme CSS, set all non blog pages to `Status: unlisted`, and rebuilt the navigation with _header.md_ in the shared folder. Feed output is as desired.

**Comment by schulle4u** [comment]
Could you share the exact problem that needs to be solved and why the existing options are not sufficient enough? Maybe we have a less painful solution and can improve the feed extension at the end. 🙂 To exclude pages from the feed a page setting would be imagenable, e.g. `Feed: exclude`. If you just need a special blog feed link for auto discovery in the header layout, this is a simple line of code. 

```
<link rel="alternate" type="application/rss+xml" href="<?php echo $this->yellow->page->getBase(true)."/feed/folder:blog/page:feed.xml" ?>" title="<?php echo "Blog - ".$this->yellow->page->getHtml("sitename") ?>" />
```

I liked the old layout filter setting, but I'm also OK with a feed for all pages, although it makes less sense for non-chronological pages. And having publications from 1970 looks odd in my feed reader. 😁

**Reply by Sussshhhh** [reply]
Thank you for your time and patience. 
Sorry, if I wasn't clear enough about what my question was. I will try to explain it in more detail;

1. I want to use Yellow mainly as a blog. I put the blog on the home page by naming a folder _1-blog_, which contains the respective _page.md_ and _New-post.md_. I am using the Paris theme. Posting to the blog works fine.
2. I also have folders for other pages (_2-about_, _3-contact_, etc.). 
3. As you correctly interpreted, I want the unfiltered feed to list only blog posts (and therefore exclude the other pages).

In the doc of the feed extension is mentioned:

> If you don't want that a page is visible, set `Status: unlisted`

I did that on the "not-blog post" pages.

The result is, the unfiltered feed only contains blog posts (and excludes the `Status: unlisted` pages).

I disabled the regular navigation in CSS by `display:none` and rebuilt it with a _header.md_ in the _shared_ folder, in order to show unlisted pages in a menu.

---

I just tried out your suggested solution with the additional code in the header layout; though the blog feed was correctly discovered in a feed reader, no blog posts appeared.

Thank you again and I hope my description was sufficient. 

_Edit:
After a clean install I noticed your suggestion works. And didn't in my previous case most likely because I deleted the `Title: Home` containing page.md. So maybe I should have asked another question first; how to put the blog on the home page (_AND_ how to make the auto discoverable feed only contain blog posts)?_

**Comment by schulle4u** [comment, marked as answer]
Thanks for letting us know about your working environment. The [feed extension](https://github.com/annaesvensson/yellow-feed) has been updated and now includes a more flexible layout filter. To make a blog feed, open `system/extensions/yellow-system.ini` and change `FeedRecentChanges: blog`. More examples how to use the layout filter can be found in the documentation. I hope this solves your problem. 🙂

**Reply by pftnhr** [reply]
Awesome, thank you!

**Reply by Sussshhhh** [reply]
This is great, @schulle4u! I appreciate your dedication regarding this issue; thanks! 💻✨

**Reply by Sussshhhh** [reply]
Thank you @annaesvensson & @markseuffert as well! 👏
